src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/DataContractSerializer.cs

```diff
@@ -82,7 +83,7 @@ public DataContractSerializer(Type type, XmlDictionaryString rootName, XmlDictio
         {
         }
 
-        public DataContractSerializer(Type type, XmlDictionaryString rootName, XmlDictionaryString rootNamespace, IEnumerable<Type> knownTypes)
+        public DataContractSerializer(Type type, XmlDictionaryString rootName, XmlDictionaryString rootNamespace, IEnumerable<Type>? knownTypes)
```

Comment:
should rootName and rootNamespace be nullable (here and anywhere above)

Reply:
The current code throws when you pass in `null` for the rootName and rootNamespace string overloads:

![image](https://user-images.githubusercontent.com/8291187/91566337-1dabd780-e909-11ea-88bf-4ef31c3b2ee9.png)

The `Initialize` method has what looks like a TODO comment:

https://github.com/dotnet/runtime/blob/84340ed4cf32d9ed75943625f2000f7fed29e230/src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/DataContractSerializer.cs#L145-L147

The only place that can pass `null` into the Initialize method is the overload that takes a `DataContractSerializerSettings`. And if you set `rootName` without setting `rootNamespace`, you can get a NRE: https://github.com/dotnet/runtime/issues/41395.

So I don't believe these should be annotated as nullable. If you are using these overloads, you should pass in valid objects.